src/pages/index.js

```diff
@@ -0,0 +1,361 @@
+      closeModal(avatarModal);
+    })
+
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Saving...", "Save");
+    });
+}
+
+//ToDo: call the addCard method from the api class
+function handleAddCardSubmit(evt) {
+  evt.preventDefault(); // Prevent the default form submission behavior
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Saving...", "Save");
+  const inputValues = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      cardForm.reset(); // Reset the form fields
+      disableBtn(cardSubmitBtn, validationConfig); // Disable the submit button
+      closeModal(cardModal); // Close the modal
+    })
+    .catch((err) => {
+      console.error(err); // Handle any errors that occur during the API call
+    })
+    .finally(() => {
+      setButtonText(submitBtn, false, "Saving...", "Save");
+    });
+}
+
+// ToDo: implement loading text for all other form submissions
+
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  // change text content to "Saving..." using setButtonText
+  const submitBtn = evt.submitter; // Get the submit button
+  setButtonText(submitBtn, true, "Saving...", "Save");
+
+  api
+    .editUserInfo({
+      name: nameInputElement.value,
+      about: descriptionInputElement.value,
+    })
+    .then((data) => {
+      //ToDo use dtd argument instead of the input values
+      nameElement.textContent = data.name;
+      descriptionElement.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // use setButtonText to reset the button text
+      setButtonText(submitBtn, false, "Saving...", "Save");
+    });
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Deleting...", "Delete");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove(); // Remove the card element from the DOM
+      closeModal(deleteModalBtn); // Close the delete confirmation modal
+    })
+    .catch(console.error)
+    .finally(() => {
+      // use setButtonText to reset the button text
+      setButtonText(submitBtn, false, "Deleting...", "Delete");
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  // Open the delete confirmation modal
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModalBtn);
+}
+
+profileEditButton.addEventListener("click", () => {
+  console.log(profileEditButton);
+  nameInputElement.value = nameElement.textContent;
+  descriptionInputElement.value = descriptionElement.textContent;
+  resetValidation(
+    editFormElement,
+    [nameInputElement, descriptionInputElement],
+    validationConfig
+  );
+  openModal(editProfileModal);
+});
+
```

Comment:
* I can't close the `confirmation` modal by clicking on the `cancel` and `close` buttons https://snipboard.io/pS1qky.jpg 
* I can't close the `card modal` by clicking on its `close icon`